doc/source/whatsnew/v2.0.0.rst

```diff
@@ -1210,12 +1210,14 @@ Datetimelike
 - Bug in :func:`to_datetime` with both ``unit`` and ``origin`` specified returning incorrect results (:issue:`42624`)
 - Bug in :meth:`GroupBy.quantile` with datetime or timedelta dtypes giving incorrect results for groups containing ``NaT`` (:issue:`51373`)
 - Bug in :meth:`Groupby.quantile` incorrectly raising with :class:`PeriodDtype` or :class:`DatetimeTZDtype` (:issue:`51373`)
+- Bug in :meth:`Timestamp.round` with values close to the implementation bounds returning incorrect results instead of raising ``OutOfBoundsDatetime`` (:issue:`51494`)
```

Comment:
Does this impact the new non-nano feature? Otherwise might be good for 2.1 instead

Reply:
Nothing specific to non-nano here, this will affect any reso

Reply:
Got it. I think it may be safer to not backport for now and move to 2.1 unless you have a strong opinion for this to be in 2.0